contrib/k8s.io/client-go/kubernetes/kubernetes.go

```diff
@@ -17,9 +18,8 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
 )
 
-const (
-	prefixAPI   = "/api/v1/"
-	prefixWatch = "watch/"
+var (
+	apiRegexp = regexp.MustCompile("^(?:/api/v1/|/apis/([a-z0-9.-]+/)([a-z0-9.-]+/))(watch/)?(namespaces/[a-z0-9.-]+/)?([a-z0-9.-]+)(/[a-z0-9.-]+)?(/[a-z0-9.-]+)?(/[a-z0-9.-]+)?$")
```

Comment:
While a regex feels like the right way to approach this problem it also significantly increases the complexity of the code here. I'm wondering if maybe we can refactor usage of this regex into a single private (and individually tested) function that is responsible just for parsing and spitting out a struct of filled in fields? Curious on if @knusbaum has any thoughts on this

Reply:
I spoke with the team a bit and on further thought this approach should be fine!